crates/pixi_manifest/src/toml/document.rs

```diff
@@ -252,4 +293,114 @@ channels = ["dummy-channel"]
         // No empty table is being created
         assert!(!manifest.0.to_string().contains("[test]"));
     }
+
+    #[derive(Serialize)]
+    struct DummyConfig {
+        version: String,
+        enabled: bool,
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        optional: Option<String>,
+    }
+    impl DummyConfig {
+        fn as_value(&self) -> toml_edit::Value {
+            Serialize::serialize(self, toml_edit::ser::ValueSerializer::new()).unwrap()
+        }
+    }
+
```

Comment:
Could you please add a test for a weird table name? E.g. with a dot inside?
No worries if it doesn't work, but we could open an issue for that

Reply:
Good idea